packages/components/src/internal/components/picklist/actions.ts

```diff
@@ -248,35 +248,39 @@ export function getSelectedPicklistSamples(
     });
 }
 
-export function getSamplesNotInList(listName: string, selectionKey?: string, sampleIds?: string[]): Promise<string[]> {
-    return new Promise((resolve, reject) => {
-        getPicklistSamples(listName)
-            .then(existingSamples => {
-                if (sampleIds) {
-                    resolve(sampleIds.filter(id => !existingSamples.has(id.toString())));
-                } else if (selectionKey) {
-                    getSelected(selectionKey).then(response => {
-                        resolve(response.selected.filter(id => !existingSamples.has(id.toString())));
-                    });
-                } else {
-                    resolve([]);
+export function getSamplesNotInList(listName: string, selectionKey?: string, useSnapshotSelection?: boolean, sampleIds?: string[]): Promise<string[]> {
+    return new Promise(async (resolve, reject) => {
+        try {
+            const existingSamples = await getPicklistSamples(listName)
+            if (sampleIds) {
+                resolve(sampleIds.filter(id => !existingSamples.has(id.toString())));
+            } else if (selectionKey) {
+                let response;
+                if (useSnapshotSelection) {
+                   response =  await getSnapshotSelections(selectionKey)
                 }
-            })
-            .catch(reason => {
-                console.error(reason);
-                reject(reason);
-            });
+                else {
+                    response = await getSelected(selectionKey)
+                }
```

Comment:
minor: this might not be worth it, but I think this type of if/else for `useSnapshotSelection` occurs in a few places. Would it make sense to just always call `response = await getSelected(selectionKey, useSnapshotSelection)`, i.e. add the useSnapshotSelection param to getSelected, and then have getSelected be the one that conditionally uses  `getSnapshotSelections` instead?

Reply:
I was torn on that since it's only in a couple of places, but it does make it cleaner in more places than uglier (and might make us think about whether snapshot selections are needed), so I did this. Thanks.